factory/config.go

```diff
@@ -51,6 +52,10 @@ type clusterConfig struct {
 	Arch             string
 }
 
+type extraConfig struct {
```

Comment:
lets give a better name please
extra config i am still in blank dont know what that means.
like name it for the purpose, networkwhateverconfig,  or  even generalConfig at least it means that we have other configs not direclty realted to the cluster.